### PR TITLE
pkg: fix bugs that the report interval equals the avg interval

### LINKED ISSUE #6301: `AvgOverTime` can not work well
### What did you do?
```
func TestPulse2(t *testing.T) {
	t.Parallel()
	re := require.New(t)
	dur := 5 * time.Second
	aot := NewAvgOverTime(dur)
	aot.Add(1000, dur)
	re.Equal(float64(1000), aot.GetInstantaneous())
	aot.Clear()
	aot.Add(1000, dur)
	re.Equal(float64(1000), aot.GetInstantaneous())
}
```

### What did you expect to see?
`GetInstantaneous` is lastest value
### What did you see instead?
`GetInstantaneous` is zero
### What version of PD are you using (`pd-server -V`)?
master

### PR DESCRIPTION
**Pull request by bufferflies**
### What problem does this PR solve?

Issue Number: Close #6301

### What is changed and how does it work?
the equal always be true if the report interval is equal the avg interval
https://github.com/tikv/pd/blob/ce053eb3201fb62942627e3e014bd274d8ff8922/pkg/movingaverage/avg_over_time.go#L87-L91

```commit-message
```

### Check List

Tests

- Unit test

Code changes

Side effects

Related changes

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/movingaverage/avg_over_time.go [hunk 1]
```diff
@@ -135,7 +136,7 @@ func (aot *AvgOverTime) GetIntervalSum() time.Duration {
 // GetInstantaneous returns the value just added.
 func (aot *AvgOverTime) GetInstantaneous() float64 {
 	if aot.que.Len() == 0 || aot.que.Back() == nil {
-		return 0
+		return aot.margin.delta
```
Comment (lhy1024) [inline pkg/movingaverage/avg_over_time.go:139]:
```suggestion
		return aot.margin.delta/aot.margin.interval
```

File: pkg/movingaverage/avg_over_time_test.go [hunk 2]
```diff
@@ -42,6 +42,19 @@ func TestPulse(t *testing.T) {
 	}
 }
 
+func TestPulse2(t *testing.T) {
+	t.Parallel()
+	re := require.New(t)
+	dur := 5 * time.Second
+	aot := NewAvgOverTime(dur)
+	aot.Add(1000, dur)
+	re.Equal(float64(1000), aot.GetInstantaneous())
```
Comment (lhy1024) [inline pkg/movingaverage/avg_over_time_test.go:51]:
```suggestion
	re.Equal(float64(1000)/dur, aot.GetInstantaneous())
```

File: pkg/movingaverage/avg_over_time_test.go [hunk 3]
```diff
@@ -42,6 +42,19 @@ func TestPulse(t *testing.T) {
...
+	re.True(aot.IsFull())
+	aot.Clear()
+	aot.Add(1000, dur)
+	re.Equal(float64(1000), aot.GetInstantaneous())
```
Comment (lhy1024) [inline pkg/movingaverage/avg_over_time_test.go:55]: ditto

**Review by lhy1024 — CHANGES_REQUESTED** [review]
aot.GetInstantaneous() should also return rate.

File: pkg/movingaverage/avg_over_time.go (hunk 1 above)
Comment (bufferflies) [reply pkg/movingaverage/avg_over_time.go:147]: fixed

File: pkg/movingaverage/avg_over_time_test.go (hunk 3 above)
Comment (bufferflies) [reply pkg/movingaverage/avg_over_time_test.go]: fixed

**Review by lhy1024 — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**